onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from onnxscript import ir
+from onnxscript.version_converter._adapter_lib import pick_adapter_set
+
+
+class _VersionConverter:
+    def __init__(self, model: ir.Model) -> None:
+        self._functions = model.functions
+        self.model_version = model.opset_imports.get("")
+        self.custom_adapters = {}
+
+    def _register_custom_adapters(self, custom_adapter_list) -> None:
+        self.custom_adapters = custom_adapter_list
+
+    def graph_version_convert(self, graph: ir.Graph, target_version: int) -> None:
+        if self.model_version == target_version:
```

Comment:
(Extension) I think we will need to soon support the case where the incoming model has nodes with different opset versions. At that point, such checks should happen at the node level, not at the model level.